web/src/pages/billing/BillingCostByCategory.tsx

```diff
@@ -0,0 +1,293 @@
+        setStart(start_update)
+        setEnd(end_update)
+        updateNav(
+            groupBy,
+            selectedGroup,
+            selectedCostCategory,
+            selectedPeriod,
+            start_update,
+            end_update
+        )
+    }
+
+    const getData = (query: BillingTotalCostQueryModel) => {
+        setIsLoading(true)
+        setError(undefined)
+        new BillingApi()
+            .getTotalCost(query)
+            .then((response) => {
+                setIsLoading(false)
+
+                // calc totals per sku
+                const recTotals: { [key: string]: number } = {}
+                response.data.forEach((item: BillingTotalCostRecord) => {
+                    const { sku, cost } = item
+                    if (!recTotals[sku]) {
+                        recTotals[sku] = 0
+                    }
+                    recTotals[sku] += cost
+                })
+                const sortedRecTotals: { [key: string]: number } = Object.fromEntries(
+                    Object.entries(recTotals).sort(([, a], [, b]) => b - a)
+                )
+                const rec_grps = Object.keys(sortedRecTotals)
+                const records: { [key: string]: { [key: string]: number } } = {}
+
+                response.data.forEach((item: any) => {
+                    const { day, sku, cost } = item
+                    if (day !== undefined) {
+                        if (!records[day]) {
+                            // initialise day structure
+                            records[day] = {}
+                            rec_grps.forEach((k: string) => {
+                                records[day][k] = 0
+                            })
+                        }
+                        records[day][sku] = cost
+                    }
+                })
+                setData(
+                    Object.keys(records).map((key) => ({
+                        date: new Date(key),
+                        values: records[key],
+                    }))
+                )
+            })
+            .catch((er) => setError(er.message))
+    }
+
+    React.useEffect(() => {
+        // if selectedCostCategory is all
+        const selFilters: { [key: string]: string } = {}
+
+        if (groupBy && selectedGroup && !selectedGroup.startsWith('All ')) {
+            selFilters[groupBy] = selectedGroup
+        }
+        if (selectedCostCategory && !selectedCostCategory.startsWith('All ')) {
+            selFilters.cost_category = selectedCostCategory
+        }
+
+        if (selectedPeriod !== undefined && selectedPeriod !== '' && selectedPeriod !== null) {
+            getData({
+                fields: [BillingColumn.Sku],
+                start_date: start,
+                end_date: end,
+                filters: selFilters,
+                order_by: { day: false },
+                time_periods: selectedPeriod,
+                // show only records with cost > 0.01
+                min_cost: 0.01,
```

Comment:
This may inadvertently remove some costs that are very low, can it be 0?

Reply:
Well, what I have noticed the slack reporting often ignore the very small values. In this case it is sum per categories,  so we want to ignore them otherwise you ended up with a lot of 0 series in the charts.
For reports on individual batch this condition is not present as we want to see all there.